graylog2-shared/src/main/java/org/graylog2/shared/buffers/JournallingMessageHandler.java

```diff
@@ -58,14 +62,30 @@ public void onEvent(RawMessageEvent event, long sequence, boolean endOfBatch) th
 
             final Converter converter = new Converter();
             // copy to avoid re-running this all the time
-            final List<Journal.Entry> entries = Lists.newArrayList(transform(batch, converter));
-            final long lastOffset = journal.write(entries);
-            log.debug("Processed batch, wrote {} bytes, last journal offset: {}, signalling reader.",
-                      converter.getBytesWritten(),
-                      lastOffset);
-            journalFilled.release();
+            // Remove all null values returned from the converter (might happen if the Converter throws an exception)
+            final List<Journal.Entry> entries = Lists.newArrayList(filter(transform(batch, converter), notNull()));
 
+            // Clear the batch list after transforming it with the Converter because the fields of the RawMessageEvent
+            // objects in there have been set to null and cannot be used anymore.
             batch.clear();
+
+            // Catch all exceptions that might happen during the journal write and retry the operation.
+            // This basically blocks if the journal write always throws an exception. Once the write succeeds, we will
+            // continue.
+            boolean done = false;
+            do {
+                try {
+                    final long lastOffset = journal.write(entries);
+                    log.debug("Processed batch, wrote {} bytes, last journal offset: {}, signalling reader.",
+                            converter.getBytesWritten(),
+                            lastOffset);
+                    journalFilled.release();
+                    done = true;
+                } catch (Exception e) {
+                    log.error("Unable to write to journal - retrying after 250ms", e);
```

Comment:
There should be an upper limit on how often we retry to write to the journal. Otherwise users will see ~4 error messages per second forever if something is wrong with the journal which will clog the Graylog log output.